federation/pkg/federatedtypes/crudtester/crudtester.go

```diff
@@ -165,25 +172,34 @@ func (c *FederatedTypeCRUDTester) CheckDelete(obj pkgruntime.Object, orphanDepen
 
 // CheckPropagation checks propagation for the crud tester's clients
 func (c *FederatedTypeCRUDTester) CheckPropagation(obj pkgruntime.Object) {
-	c.CheckPropagationForClients(obj, c.clusterClients)
+	c.CheckPropagationForClients(obj, c.clusterClients, true)
 }
 
 // CheckPropagationForClients checks propagation for the provided clients
-func (c *FederatedTypeCRUDTester) CheckPropagationForClients(obj pkgruntime.Object, clusterClients []clientset.Interface) {
+func (c *FederatedTypeCRUDTester) CheckPropagationForClients(obj pkgruntime.Object, clusterClients []clientset.Interface, objExpected bool) {
 	namespacedName := c.adapter.NamespacedName(obj)
 
+	waitTimeout := wait.ForeverTestTimeout
+	if !objExpected {
+		// Setting to a reasonably lower time to reduce test times
+		waitTimeout = ObjectNotExpectedWaitTime
+	}
+
 	c.tl.Logf("Waiting for %s %q in %d clusters", c.kind, namespacedName, len(clusterClients))
 	for _, client := range clusterClients {
-		err := c.waitForResource(client, obj)
-		if err != nil {
+		err := c.waitForResource(client, obj, waitTimeout)
+		if err != nil && objExpected {
```

Comment:
I think it would be a good idea to provide a different failure for the case of an object being expected but not being found.  The following logic would also ensure that the test would fail if objExpected == false and the error is not due to timeout, which isn't true of the proposed logic.

```golang
switch {
case err == wait.ErrWaitTimeout:
  if objExpected {
   ...
  }
case err != nil:
 ...
case err == nil && !objExpected:
 ...
}
```